staging/src/k8s.io/controller-manager/config/v1beta1/types.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
```

Comment:
Are we getting rid of GenericControllerManagerConfiguration (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/controller-manager/config/v1alpha1/types.go#L25)? If not then I think it should be moving to v1beta1 as well.

Reply:
`GenericControllerManagerConfiguration` is extracted from common part of corresponding structures of KCM and CCM during the "make staging/controller-manager easier to consume task by @cici37". It is still needed unless we want to rollback any changes during the task. However, `GenericControllerManagerConfiguration` should not go to `v1beta` because:

- The stability is unknown, but it is definitely less than beta
- We are keeping `v1alpha` package following convention of how we graduate resources with feature, it would not hurt to keep GenericControllerManagerConfiguration there.
-  If we are to graduate GenericControllerManagerConfiguration to beta, it would be far out of scope of this PR. We can do this in a separate PR to avoid scope creeping.

Reply:
What does it mean to say LeaderMigrationConfiguration has gone to beta, when its not a K8s resource and is only used from GenericControllerManagerConfiguration which if we don;t move it, is obviously only going to use the v1alpha1 version of LeaderMigrationConfiguration?